src/Magnum/Trade/PhongMaterialData.h

```diff
@@ -124,6 +124,49 @@ class MAGNUM_TRADE_EXPORT PhongMaterialData: public AbstractMaterialData {
 
+        /**
+         * @brief Constructor with texture layers
+         * @param flags                Material flags
+         * @param ambientColor         Ambient color. Use
+         *      @cpp 0x000000ff_rgbaf @ce for a default value for a
+         *      non-textured material and @cpp 0xffffffff_rgbaf @ce for a
+         *      default value for a textured material.
+         * @param ambientTexture       Ambient texture ID. Ignored if @p flags
+         *      doesn't have @ref Flag::AmbientTexture
+         * @param ambientTextureLayer  Ambient texture layer. Ignored if @p flags
+         *      doesn't have @ref Flag::AmbientTexture
+         * @param diffuseColor         Diffuse color. Use
+         *      @cpp 0xffffffff_rgbaf @ce for a default value for both a
+         *      non-textured and a textured material.
+         * @param diffuseTexture       Diffuse texture ID. Ignored if @p flags
+         *      doesn't have @ref Flag::DiffuseTexture
+         * @param diffuseTextureLayer  Diffuse texture layer. Ignored if @p flags
+         *      doesn't have @ref Flag::DiffuseTexture
+         * @param specularColor        Specular color. Use
+         *      @cpp 0xffffffff_rgbaf @ce for a default value for both a
+         *      non-textured and a textured material.
+         * @param specularTexture      Specular texture ID. Ignored if @p flags
+         *      doesn't have @ref Flag::SpecularTexture.
+         * @param specularTextureLayer Specular texture layer. Ignored if @p flags
+         *      doesn't have @ref Flag::SpecularTexture.
+         * @param normalTexture        Normal texture ID. Ignored if @p flags
+         *      doesn't have @ref Flag::NormalTexture.
+         * @param normalTextureLayer   Normal texture layer. Ignored if @p flags
+         *      doesn't have @ref Flag::NormalTexture.
+         * @param textureMatrix        Texture coordinate transformation. Ignored
+         *      if @p flags doesn't have @ref Flag::TextureTransformation.
+         * @param alphaMode            Alpha mode. Use
+         *      @ref MaterialAlphaMode::Opaque for a default value.
+         * @param alphaMask            Alpha mask value. Use @cpp 0.5f @ce for a
+         *      default value.
+         * @param shininess            Shininess. Use @cpp 80.0f @ce for a default
+         *      value.
+         * @param importerState        Importer-specific state
+         * @m_since_latest
+         */
+        explicit PhongMaterialData(Flags flags, const Color4& ambientColor, UnsignedInt ambientTexture, UnsignedInt ambientTextureLayer, const Color4& diffuseColor, UnsignedInt diffuseTexture, UnsignedInt diffuseTextureLayer, const Color4& specularColor, UnsignedInt specularTexture, UnsignedInt specularTextureLayer, UnsignedInt normalTexture, UnsignedInt normalTextureLayer, const Matrix3& textureMatrix, MaterialAlphaMode alphaMode, Float alphaMask, Float shininess, const void* importerState = nullptr) noexcept;
```

Comment:
Because custom coordinate sets need explicit support in shaders the same way as texture matrix needs explicit support in a shader, I'm thinking that this should be an extra flag -- for example

```cpp
            /**
             * The material uses non-default texture coordinate sets
             * @m_since_latest
             */
            TextureCoordinateSets = 1 << 6
```

and then in the constructor asserting that non-zero sets are used only if this flag is present (hmm, the textureMatrix it's ignored if the corresponding flag is not there but I think an assert is better. Would be great if you could change that to an assert also.).

Besides that, there's currently no builtin support for this anywhere, not even in `compile()`. Don't you need that also? Back when I did attribute allocation I [reserved some space for additional texture coordinates](https://doc.magnum.graphics/magnum/structMagnum_1_1Shaders_1_1Generic.html), but those still need to be made real and then used in `compile()`. Without that, this PR has very little use on its own :)

Actually, how's the mapping meant to be done on the shader side? The shader gets a mesh with (let's say) 3 sets of texture coordinates, how can it know which is which? How do you do this in your shaders?